src/modules/Servicehosting/Service.php

```diff
@@ -552,6 +552,29 @@ public function toHostingServerApiArray(\Model_ServiceHostingServer $model, $dee
         return $result;
     }
 
+    public function toHostingAccountApiArray(\Model_ServiceHosting $model, $deep = false, $identity = null): array
+    {
+        $result = [
+            'id' => $model->id,
+            'sld' => $model->sld,
+            'tld' => $model->tld,
+            'client_id' => $model->client_id,
+            'server_id' => $model->service_hosting_server_id,
+            'plan_id' => $model->service_hosting_hp_id,
+            'reseller' => $model->reseller,
+        ];
+
+        if ($identity instanceof \Model_Admin) {
+            $result['ip'] = $model->ip;
+            $result['username'] = $model->username;
+            $result['password'] = $model->pass;
```

Comment:
If I'm not mistaken, the password field was deprecated and only records a fixed string instead of the actual password, so there wouldn't be any value in returning that field